### DISCUSSION TITLE
I'm giving a talk on MicroPython

### DISCUSSION THREAD

**Discussion by obdevel** [micropython/micropython#11958] [General]
I'm giving a talk on MicroPython to my local PUG in Cambridge, UK in a few weeks.

They've never had a talk on this subject before. What's the one thing I should not leave out ?

(And yeah, I'll be using a Pico for the demos :) )

**Comment by ricksorensen** [comment]
Interrupt handline

**Comment by jimmo** [comment]
If it's a PUG then hardware specific stuff might not be quite the same impact as it would for a hardware/electronics group. i.e. my normal quick demo is using the REPL to discover board functionality / probe i2c devices / toggle GPIO, which is only really impressive if you've used Arduino or other C-based vendor tools.

In general when I demo MicroPython to Python people they need a bit of convincing that it is "actually Python" so maybe showing language features like f-strings, classes, comprehensions, async/await, etc and standard library features like json, zlib, asyncio.

One of my absolutely favourite things to demo is `mpremote mount` and generally the way filesystems (and the VFS) are handled in MicroPython.

**Reply by scruss** [reply]
I wouldn't leave out physical computing entirely, though. It's reassuring to have MicroPython around to do embedded stuff in a familiar language. What passes for C in the PIC/AVR community would scare most programmers away, because it's mostly setting bit-masks according to values stored in huge datasheets.

**Comment by obdevel** [comment]
Thank you for your thoughts. I think some people have tinkered with Arduino and similar. It's an online meeting so I will setup a separate camera to show some visually appealing stuff (LED matrix) as well as some switches and LEDs to show asyncio/interrupts and aiorepl (which I think will impress), and talk generally about how embedded dev differs. It's a fairly tech audience, just not in embedded.

How do I answer the potential question: "where is micropython being used for real world stuff, not just hobbyist projects ?". Are there any examples I can talk about ?

Thanks again.

**Reply by mattytrentini** [reply]
You can allude to the company I work for, [Planet Innovation](https://planetinnovation.com/). We have used MicroPython on multiple projects for development of medical devices.

**Reply by arbobeck** [reply]
How about this talk from Christine Spindler called "How MicroPython went into space": https://www.youtube.com/watch?v=Vh_5Lz1mLq8

She talks about both [ESA](https://essr.esa.int/project/micropython-for-leon-pre-qualified-version) and [Spacebel](https://www.spacebel.com/)

**Comment by bidrohini** [comment]
Here is a slide on the use of Micropython in IoT. This may give you some references: 

https://www.slideshare.net/jacquessupcik/micropython-for-the-iot

Since, you'll be featuring a Raspberry Pi Pico, you must talk about the Raspberry Pi Pico Dual Core Programming with MicroPython

https://www.theengineeringprojects.com/2022/05/raspberry-pi-pico-dual-core-programming-with-micropython.html

**Comment by scruss** [comment]
> Since, you'll be featuring a Raspberry Pi Pico, you must talk about the Raspberry Pi Pico Dual Core Programming with MicroPython

I'd give that a hard swerve myself. This is an intro talk, you don't want them horrified and demoralized from the get-go. As the old joke goes: "Programming problem? Try threads: now two you problems! have"
